azurerm/internal/services/datafactory/data_factory_resource.go

```diff
@@ -421,12 +484,21 @@ func flattenDataFactoryIdentity(identity *datafactory.FactoryIdentity) interface
 	if identity.TenantID != nil {
 		tenantId = identity.TenantID.String()
 	}
+	var identityIds []string
+	if identity.UserAssignedIdentities != nil {
+		userIdentities := make([]string, 0)
+		for key := range identity.UserAssignedIdentities {
+			userIdentities = append(userIdentities, key)
+		}
+		identityIds = userIdentities
+	}
```

Comment:
First we might need to simplify this part - we do not need two slices of strings here. And before we setting those IDs, we need to first parse it and then format it.
```suggestion
	var identityIds []string
	if identity.UserAssignedIdentities != nil {
		for key := range identity.UserAssignedIdentities {
		        id, err := msiparse.UserAssignedIdentityID(key)
		        if err != nil {
		              return nil, err
		        }
			identityIds = append(identityIds , id.ID())
		}
	}
```
In this way we can keep the casing of user assigned identity consistent

Reply:
also since the parsing function will return an error, you might need to change your function signature to make sure this function could return an error